File: 3248-snake-in-matrix/3248-snake-in-matrix.java
class Solution {
    public int finalPositionOfSnake(int n, List<String> commands) {
        int ans = 0;
        for(String c : commands){
            if(c.startsWith("R")){
                ans++;
            }else if(c.startsWith("L")){
                ans--;
            }else if(c.startsWith("U")){
                ans -= n;
            }else{
                ans += n;
            }
        }
        return ans;
    }
}

File: 3248-snake-in-matrix/3248-snake-in-matrix.cpp
class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands) {
        int i=0, j=0;
        for (string s : commands) {
            if (s == "LEFT") {
                j--;
            } else if (s == "RIGHT") {
                j++;
            } else if (s == "DOWN") {
                i++;
            } else if (s == "UP") {
                i--;
            }
        }
        return i*n+j;
    }
};